pkg/telemetery/settings.go

```diff
@@ -16,23 +16,23 @@ type Setting struct {
 	Logger         *zap.Logger
 	TracerProvider trace.TracerProvider
 	Metrics        metrics.Factory
-	ReportStatus   func(*component.StatusEvent)
+	ReportStatus   func(*componentstatus.Event)
```

Comment:
I believe this is the only public-facing breaking change. Is handling it like this ok?